Compile for AFL

ironage opened the pull request:
This adds a build target specifically for AFL.
- It gives a significant build speed increase because the whole test suite doesn't have to be instrumented.
- Coverage reports will be much more accurate (and higher!) by not including the test code.

I know CMake will be available soon, but the code changes would have to be done anyways and the reduction in compile time is useful to me now.

Relates to part of #1932.

jedelbo commented:
It is confusing me that this PR is targeting `js/afl-improvements` and not `develop`.

danielpovlsen commented on test/fuzz_group.cpp:
```diff
@@ -1001,6 +1010,53 @@ void parse_and_apply_instructions(std::string& in, const std::string& path, util
+        if (File::is_dir(m_path + ".management"))
+            remove_dir(m_path + ".management");
+        File::try_remove(m_path + ".lock");
+    }
```
I am not a fan of this information about locks, .management folders etc. being in multiple places. If the `SharedGroupTestPathGuard` can be reused somehow, it would be better.

ironage commented:
@danielpovlsen good point. I removed the duplicated code by refactoring our test utility to not depend on the whole test suite, then I included it into the fuzz test project.

@jedelbo I rebased to develop, hopefully that clears things up.

danielpovlsen commented on Jenkinsfile:
```diff
@@ -471,7 +471,7 @@ def doBuildAndroid(def isPublishingRun) {
         }
 
-        node('android-hub') {
+        node('fastlinux') {
```
This can be safely reverted now.
